libs/blocks/form/form.js

```diff
@@ -290,8 +290,9 @@ async function createForm(formURL, thankYou, formData) {
 }
 
 export default async function decorate(block, formData = null) {
-  const form = block.querySelector('a[href$=".json"]');
+  const [form, endpoint] = block.querySelectorAll(':scope > div:first-child a');
   const thankYou = block.querySelector(':scope > div:last-of-type > div');
   thankYou.remove();
-  if (form) form.replaceWith(await createForm(form.href, thankYou, formData));
+  if (endpoint) endpoint.parentElement.remove();
+  if (form) form.replaceWith(await createForm(form.href, thankYou, formData, endpoint?.href));
```

Comment:
4 arguments might justify transforming the config into an object to make it easier and less error-prone to define and maintain.